src/main/java/school/lemon/changerequest/java/io/hw1/IFileManagerImp.java

```diff
@@ -0,0 +1,625 @@
+package school.lemon.changerequest.java.io.hw1;
+
+
+import school.lemon.changerequest.java.io.hw1.exception.*;
+
+import java.io.*;
+import java.nio.file.Paths;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+
+
+public class IFileManagerImp implements IFileManager {
```

Comment:
Each operation that is done by the given file manager should use it's `currentDirectory` state to check whether file with given name can be in the current directory.
Pay attention on this.